crates/matrix-sdk-crypto/src/olm/utility.rs

```diff
@@ -50,56 +53,148 @@ impl SignJson for Ed25519SecretKey {
     }
 }
 
+/// A trait for objects that are able to verify signatures.
 pub trait VerifyJson {
-    /// Verify a signed JSON object.
+    /// Verify a signature over the SignedJsonObject using this public Ed25519
+    /// key.
     ///
-    /// The object must have a signatures key associated  with an object of the
-    /// form `user_id: {key_id: signature}`.
+    /// # Arguments
+    ///
+    /// * `user_id` - The user that claims to have signed this object.
+    ///
+    /// * `key_id` - The ID of the key that was used to sign this object.
+    /// **Note**: The key ID must match the ID of the public key that is
+    /// verifying the signature. This is only used to find the correct
+    /// signature.
+    ///
+    /// * `signed_object` - The signed object that we should check for a valid
+    /// signature.
     ///
     /// Returns Ok if the signature was successfully verified, otherwise an
     /// SignatureError.
+    fn verify_json(
```

Comment:
I think it would be best if this mentioned signatures somehow, to make it clear at the call site that we're verifying a signature.

I'm even tempted to say it should be `verify_signature` instead of `verify_json`? Especially given that the third argument just accepts a trait object, so it may not even be clear to the reader that there is JSON involved.